### PR TITLE
feat: vip average tvl for preventing attack

### PR DESCRIPTION
**Pull request by djm07073**
- create new module `vip_tvl` : recommend more nice name
- feat: average tvl snapshot
- add finalize event on minita score

## Summary by CodeRabbit

- **New Features**
  - Introduced Total Value Locked (TVL) snapshots for bridges.
  - Updated terminology from 'stages' to 'epochs' in VIP weight voting processes.
  - Enhanced delegation information management for clarity.

- **Improvements**
  - Refined precision and rounding behavior in decimal multiplications for better accuracy.
  - Added `FinalizedScoreEvent` to track score updates.

- **Bug Fixes**
  - Corrected calculation issues in the `fund_reward` function.

### CODE REVIEW

File: precompile/modules/initia_stdlib/sources/decimal128.move [hunk 1]
```diff
@@ -97,10 +97,12 @@ module initia_std::decimal128 {
     }
 
     public fun div_u64(decimal: &Decimal128, val: u64): Decimal128 {
+        assert!(val != 0, EDIV_WITH_ZERO);
```
Comment (beer-1) [inline precompile/modules/initia_stdlib/sources/decimal128.move:100]: cloud also update decimal256 too?

File: precompile/modules/initia_stdlib/sources/vip/tvl_manager.move
Comment (beer-1) [inline precompile/modules/initia_stdlib/sources/vip/tvl_manager.move:1]: good to run move formatter

File: precompile/modules/initia_stdlib/sources/vip/vip.move [hunk 2]
```diff
@@ -1202,6 +1234,9 @@ module initia_std::vip {
         );
 
         module_store.stage = stage + 1;
+
+        add_tvl_snapshot_internal(module_store);
```
Comment (beer-1) [inline precompile/modules/initia_stdlib/sources/vip/vip.move:1238]: good to add comment

**Review by beer-1 — CHANGES_REQUESTED** [review]
left few comments

File: precompile/modules/initia_stdlib/sources/vip/vip.move (hunk 2 above)
Comment (Vritra4) [reply precompile/modules/initia_stdlib/sources/vip/vip.move:1238]: good call! we have to lift readability up

**Review by beer-1 — APPROVED** [review]
LGTM

**Pull request merged**